cpp/src/sampling/random_walks_impl.cuh

```diff
@@ -156,15 +202,73 @@ struct biased_selector {
     //  instead of making a decision based on the index I need to find
     //  upper_bound (or is it lower_bound) of the random number and
     //  the cumulative weight.
-    CUGRAPH_FAIL("biased sampling not implemented");
+
+    //  Create vertex frontier
+    using vertex_t = typename GraphViewType::vertex_type;
+
+    using tag_t = void;
+
+    cugraph::vertex_frontier_t<vertex_t, tag_t, GraphViewType::is_multi_gpu, false> vertex_frontier(handle, 1);
+
+    vertex_frontier.bucket(0).insert(current_vertices.begin(), current_vertices.end());
+
+    // Create data structs for results
+    rmm::device_uvector<vertex_t> minors(0, handle.get_stream());
+        // Should this be optional? Necessary for biased
+    std::optional<rmm::device_uvector<weight_t>> weights{std::nullopt};
+
+    if (edge_weight_view) {
+      auto vertex_weight_sum  = compute_out_weight_sums(handle, graph_view, *edge_weight_view);
+      edge_src_property_t<GraphViewType, weight_t> edge_src_out_weight_sums(handle, graph_view);
+      update_edge_src_property(handle, graph_view, vertex_weight_sum.data(), edge_src_out_weight_sums);
+      auto [sample_offsets, sample_e_op_results] =
+        cugraph::per_v_random_select_transform_outgoing_e(
+          handle,
+          graph_view,
+          vertex_frontier.bucket(0),
+          edge_src_out_weight_sums.view(),
+          cugraph::edge_dst_dummy_property_t{}.view(),
+          *edge_weight_view,
+          biased_random_walk_e_bias_op_t<vertex_t, weight_t>{},
+          biased_sample_edges_op_t<vertex_t, weight_t>{},
+          rng_state_,
+          size_t{1},
+          true,
+          std::make_optional(
+            thrust::make_tuple(vertex_t{cugraph::invalid_vertex_id<vertex_t>::value}, weight_t{0.0})));
+      minors = std::move(std::get<0>(sample_e_op_results));
+      weights = std::move(std::get<1>(sample_e_op_results));
+    } else {
+      //  Just uniform random walk
```

Comment:
We have an existing uniform random walk function, right?

I think we'd better make `edge_weight_view` non-optional. If the caller doesn't have a valid `edge_weight_view` object, the caller should invoke the uniform random walk function rather than calling this function with std::nullopt. No need to implement uniform random walk in two different places.